osf_tests/test_guid.py

```diff
@@ -452,3 +457,77 @@ def test_resolve_guid_download_errors(self):
 
         res = self.app.get(pp.url + 'download', auth=non_contrib.auth)
         assert res.status_code == 410
+
+@pytest.fixture()
+def creator():
+    return AuthUserFactory()
+
+@pytest.fixture()
+def auth(creator):
+    return Auth(user=creator)
+
+@pytest.fixture()
+def preprint_provider():
+    return PreprintProviderFactory()
+
+@pytest.mark.django_db
+class TestGuidVersionsThrough:
+    def test_creation_versioned_guid(self, creator, preprint_provider):
+        preprint = Preprint.create(
+            creator=creator,
+            provider=preprint_provider,
+            title='Preprint',
+            description='Abstract'
+        )
+
+        version_entry = preprint.versioned_guids.first()
+        assert preprint.creator == creator
+        assert preprint.provider == preprint_provider
+        assert preprint.title == 'Preprint'
+        assert preprint.description == 'Abstract'
+
+        assert version_entry.guid == preprint.guids.first()
+        assert version_entry.referent == preprint
+        assert version_entry.content_type.model == 'preprint'
+        assert version_entry.object_id == preprint.pk
+        assert version_entry.version == 1
+
+    def test_create_version(self, creator, preprint_provider):
+        preprint = Preprint.create(
+            creator=creator,
+            provider=preprint_provider,
+            title='Preprint',
+            description='Abstract'
+        )
+        first_preprint_guid = preprint.guids.first()
+        preprint.is_published = True
+        preprint.save()
+        auth = Auth(user=creator)
+        new_preprint, data_for_update = Preprint.create_version(
+            create_from_guid=preprint._id,
+            auth=auth
+        )
+        new_version = new_preprint.versioned_guids.first()
+
+        assert new_version.referent == new_preprint
+        assert new_version.object_id == new_preprint.pk
+        assert new_version.content_type.model == 'preprint'
+        assert new_preprint.guids.first() == first_preprint_guid
+        assert new_version.version == 2
+        assert GuidVersionsThrough.objects.count() == 2
```

Comment:
This will pass of course but the testing goal is: the preprint object now has two versions instead of `GuidVersionsThrough` table has two entries. The table could have versions from other preprints too. Let's check something like `new_preprint.guids.first().versions.count()`.

Reply:
Changed this check